crates/task-impls/src/consensus/mod.rs

```diff
@@ -385,186 +381,11 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusTaskState<TYPES, I>
-                    let old_anchor_view = consensus.last_decided_view;
-                    consensus.collect_garbage(old_anchor_view, new_anchor_view);
-                    consensus.last_decided_view = new_anchor_view;
-                    consensus
-                        .metrics
-                        .last_decided_time
-                        .set(Utc::now().timestamp().try_into().unwrap());
-                    consensus.metrics.invalid_qc.set(0);
-                    consensus
-                        .metrics
-                        .last_decided_view
-                        .set(usize::try_from(consensus.last_decided_view.get_u64()).unwrap());
-                    let cur_number_of_views_per_decide_event =
-                        *self.cur_view - consensus.last_decided_view.get_u64();
-                    consensus
-                        .metrics
-                        .number_of_views_per_decide_event
-                        .add_point(cur_number_of_views_per_decide_event as f64);
-
-                    debug!("Sending Decide for view {:?}", consensus.last_decided_view);
-                    debug!("Decided txns len {:?}", included_txns_set.len());
-                    decide_sent.await;
-                    debug!("decide send succeeded");
-                }
-
-                let new_view = self.current_proposal.clone().unwrap().view_number + 1;
-                // In future we can use the mempool model where we fetch the proposal if we don't have it, instead of having to wait for it here
-                // This is for the case where we form a QC but have not yet seen the previous proposal ourselves
-                let should_propose = self.quorum_membership.get_leader(new_view) == self.public_key
-                    && consensus.high_qc.view_number
-                        == self.current_proposal.clone().unwrap().view_number;
-                // todo get rid of this clone
-                let qc = consensus.high_qc.clone();
-
-                drop(consensus);
-                if new_decide_reached {
-                    self.cancel_tasks(new_anchor_view).await;
-                }
-                if should_propose {
-                    debug!(
-                        "Attempting to publish proposal after voting; now in view: {}",
-                        *new_view
-                    );
-                    if let Err(e) = self
-                        .publish_proposal(qc.view_number + 1, event_stream.clone())
-                        .await
-                    {
-                        warn!("Failed to propose; error = {e:?}");
-                    };
-                }
-
-                if !self.vote_if_able(&event_stream).await {
-                    return;
+                if let Err(e) =
+                    handle_quorum_proposal_validated(proposal, event_stream.clone(), self).await
+                {
+                    warn!(?e, "Failed to handle QuorumProposalValidated event");
```

Comment:
I think this should be `debug` or `info` at best, not `warn` since the function calls `vote_if_able`, which can fail for multiple reasons (some benign).

Reply:
Also, I think `?e` uses the `Debug` formatter which prints a backtrace (which I don't think we want here, that's what made the `update_view` log really verbose last time)

I think `warn!("Failed to handle ...: {e}");` might avoid that, but I'd have to test (can check later tonight)

Reply:
No problem, let me check this one